src/index.ts

```diff
@@ -44,25 +60,124 @@ export class WalletPluginMetaMask extends AbstractWalletPlugin implements Wallet
         })
     }
 
+    async lookupAccounts(publicKey: PublicKey, chainId: Checksum256Type): Promise<AccountFound[]> {
+        const response = await fetch(
+            `${ACCOUNT_LOOKUP_URL}/lookup/${publicKey.toLegacyString()}?includeTestnets=true`
+        )
+        const accountsByNetwork: AccountLookup[] = await response.json()
+
+        const networkAccount = accountsByNetwork.find(
+            (networkAccount: AccountLookup) => networkAccount.chainId === String(chainId)
+        )
+
+        if (!networkAccount) {
+            return []
+        }
+
+        return networkAccount.accounts
+    }
+
+    async getPermissionLevel(
+        context: LoginContext,
+        chain: ChainDefinition
+    ): Promise<PermissionLevel> {
+        if (context.permissionLevel) {
+            return context.permissionLevel
+        }
+
+        const activeKey = await this.retrievePublicKey(chain.id, 1)
+        const accounts = await this.lookupAccounts(activeKey, chain.id)
+
+        if (!context.ui) {
+            throw new Error('UI not found')
+        }
+
+        return new Promise((resolve) => {
+            const createAccount = async () => {
+                // Changed to arrow function
+                const qs = new URLSearchParams()
+                qs.set('supported_chains', String(chain))
+                if (context.appName) {
+                    qs.set('scope', String(context.appName))
+                }
+
+                const ownerKey = await this.retrievePublicKey(chain.id, 0)
+
+                qs.set('owner_key', String(ownerKey))
+                qs.set('active_key', String(activeKey))
+                const accountCreator = new AccountCreator({
+                    supportedChains: [String(chain.id)],
+                    fullCreationServiceUrl: `${ACCOUNT_CREATION_SERVICE_URL}?${qs.toString()}`,
+                    scope: context.appName || 'Antelope App',
+                })
+                accountCreator.createAccount().then((accountCreationResponse) => {
+                    if ('sa' in accountCreationResponse) {
+                        resolve(
+                            PermissionLevel.from({
+                                actor: accountCreationResponse.sa,
+                                permission: accountCreationResponse.sp,
+                            })
+                        )
+                    } else {
+                        throw new Error(accountCreationResponse.error || 'Account creation failed')
+                    }
+                })
+            }
+
+            context.ui.prompt({
```

Comment:
You shouldn't have to prompt this, the web-renderer should do this 🤔

Reply:
I could be missing something, but as far as I could tell, there was no way to have web renderer prompt for an account while also giving the option to create a new account so that's why I ended with something like this.